datafusion/src/main.rs

```diff
@@ -41,11 +41,12 @@ async fn main() {
         let frame = ctx.sql(query).await.unwrap();
         let mut s = frame.execute_stream().await.unwrap();
 
-        for batch in s.next().await {
-            batch.unwrap();
+        let mut row_count = 0;
+        while let Some(batch) = s.next().await {
```

Comment:
Previously this would just take the first RecordBatch returned, i.e. the first 8192 rows.

Fortunately most queries weren't returning more than this, and so it doesn't impact most of the benchmarks, but does explain the somewhat extreme unpredicated scan performance :laughing: